java/org/apache/tomcat/util/net/AprEndpoint.java

```diff
@@ -292,52 +295,79 @@ public void bind() throws Exception {
 
         // Create the pool for the server socket
         serverSockPool = Pool.create(rootPool);
+
         // Create the APR address that will be bound
-        String addressStr = null;
-        if (getAddress() != null) {
-            addressStr = getAddress().getHostAddress();
-        }
-        int family = Socket.APR_INET;
-        if (Library.APR_HAVE_IPV6) {
-            if (addressStr == null) {
-                if (!OS.IS_BSD) {
+        if (getPath() != null) {
+            if (Library.APR_HAVE_UNIX) {
+                hostname = getPath().toString();
```

Comment:
That hurts in terms of naming...

Reply:
I can try just name if you like?

hostname matches the APR implementation.

Reply:
OK, go with APR consistency here. Though, I'd rather see `address`, but consistency is more important.